pkg/models/sigv4.go

```diff
@@ -24,6 +24,13 @@ const (
 	Credentials AuthType = "credentials"
 )
 
+var permittedHeaders = map[string]struct{}{
```

Comment:
Shouldn't `Host` be in here?

Reply:
Should have mentioned - it turns out Go uses the `req.Host` field as the source of truth for that and Go takes care of that https://github.com/golang/go/blob/cad6d1fef5147d31e94ee83934c8609d3ad150b7/src/net/http/request.go#L92. I think it makes sense to add it here still though so thanks for the reminder!

Reply:
It could be that it won't ever be in the header map (AFAICT at least) - but might be confusing to omit it. At the very least, probably should have a comment but I think putting the entry in the map is fine